models/spike_train_injector.h

```diff
@@ -0,0 +1,374 @@
+#include "event.h"
+#include "nest_time.h"
+#include "nest_types.h"
+#include "node.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, device, spike, generator
+
+Short description
++++++++++++++++++
+
+Neuron that emits prescribed spike trains.
+
+Description
++++++++++++
+
+The spike train injector neuron emits spikes at prescribed spike times
+which are given as an array. The neuron does not allow incoming connections
+and is thus not able to process incoming spikes or currents.
+
+.. note::
+
+   ``spike_train_injector`` is recommended if the spike trains have a similar
+   rate to regular neurons. For very high rates, use
+   :doc:`spike generator </models/spike_generator>`.
+
+Spike times are given in milliseconds as an array. The `spike_times` array
+must be sorted with the earliest spike first. All spike times must be strictly
+in the future. Trying to set a spike time in the past or at the current time
+step will cause a NEST error. Setting a spike time of 0.0 will also result
+in an error.
+
+Multiple occurrences of the same time indicate that more than one event is to
+be generated at this particular time.
+
+The spike generator supports spike times that do not coincide with a time
+step, i.e., are not falling on the grid defined by the simulation resolution.
+There are three options that control how spike times that do not coincide
+with a step are handled (see also examples below):
+
+    `precise_times`  default: false
+
+If false, spike times will be rounded to simulation steps, i.e., multiples
+of the resolution. The rounding is controlled by the two other flags. If true,
+spike times will not be rounded but represented exactly as a combination of
+step and offset. This should only be used if all neurons receiving the spike
+train can handle precise timing information. In this case, the other two
+options are ignored.
+
+    `allow_offgrid_times`   default: false
+
+If false, spike times will be rounded to the nearest step if they are less
+than tic/2 from the step, otherwise NEST reports an error. If true, spike
+times are rounded to the nearest step if within tic/2 from the step, otherwise
+they are rounded up to the *end* of the step. This setting has no effect if
+precise_times is true.
+
+    `shift_now_spikes`   default: false
+
+This option is mainly for use by the PyNN-NEST interface. If false, spike
+times rounded down to the current point in time will be considered in the past
+and ignored. If true, spike times that are rounded down to the current time
+step are shifted one time step into the future.
+
+Note that ``GetStatus`` will report the spike times that the spike_generator
+will actually use, i.e., for grid-based simulation the spike times rounded
+to the appropriate point on the time grid. This means that ``GetStatus`` may
+return different `spike_times` values at different resolutions.
+
+Example:
+
+  ::
+
+     nest.Create("spike_train_injector",
+                 params={"spike_times": [1.0, 2.0, 3.0]})
```

Comment:
@jessica-mitchell: can you please give this model description (especially the code blocks) a quick run-over to see if everything corresponds to our guidelines? Thanks!